From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jab125 <67534807+Jab125@users.noreply.github.com>
Date: Sat, 13 May 2023 20:20:49 +1000
Subject: [PATCH] Change maven group


diff --git a/build.gradle.kts b/build.gradle.kts
index 180cf74a29442396588a7eb2fe8ac9f561da09ff..89d5f313bcb44a1af14d50c73181a6316396ffc3 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -20,7 +20,7 @@ plugins {
     groovy
 }
 
-group = "com.github.replaymod"
+group = "com.jab125" // this is a fork
 version = "SNAPSHOT"
 
 val kotestVersion: String by project.extra
