From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jab125 <67534807+Jab125@users.noreply.github.com>
Date: Sat, 13 May 2023 21:19:05 +1000
Subject: [PATCH] Add version to manifest


diff --git a/build.gradle.kts b/build.gradle.kts
index 426712c3693a7e603227782a8803f175c07cc8c6..2ce2b8f4ef56c7eb411409b6dc2c5a9f79f09d84 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -41,6 +41,11 @@ gradlePlugin {
 java {
     sourceCompatibility = JavaVersion.VERSION_1_8
     targetCompatibility = JavaVersion.VERSION_1_8
+
+    manifest {
+        attributes["Implementation-Version"] = project.version
+    }
+
 }
 
 tasks.withType<Test> {
