From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jab125 <67534807+Jab125@users.noreply.github.com>
Date: Sat, 13 May 2023 21:19:05 +1000
Subject: [PATCH] Add version to manifest


diff --git a/build.gradle.kts b/build.gradle.kts
index ab4a8cd89935b5788fd6ae08b7ff96d60d7dd2e9..46f9ba1cf205dccf839654f6401283160ce74165 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -41,6 +41,11 @@ gradlePlugin {
 java {
     sourceCompatibility = JavaVersion.VERSION_1_8
     targetCompatibility = JavaVersion.VERSION_1_8
+
+    manifest {
+        attributes["Implementation-Version"] = project.version
+    }
+
 }
 
 tasks.withType<Test> {
