From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jab125 <67534807+Jab125@users.noreply.github.com>
Date: Tue, 23 May 2023 23:09:14 +1000
Subject: [PATCH] Ignore some errors

WILL BE REVERTED LATER

diff --git a/build.gradle.kts b/build.gradle.kts
index d35b6cb55eb2af8129203dae4ea1de51e5925b44..83ac6c913f21cf96d6b46dcbfd291363b2671f19 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -21,7 +21,7 @@ plugins {
 }
 
 group = "com.jab125.preprocessor" // this is a fork
-version = "0.1.0"
+version = "0.1.1"
 
 val kotestVersion: String by project.extra
 
diff --git a/src/main/kotlin/com/replaymod/gradle/preprocess/PreprocessPlugin.kt b/src/main/kotlin/com/replaymod/gradle/preprocess/PreprocessPlugin.kt
index 4b9d3ece7e51670f8c44cd0312bc467e9d2bf568..4cd124b919a95f0bed82604b84b0281b0c2f8c81 100644
--- a/src/main/kotlin/com/replaymod/gradle/preprocess/PreprocessPlugin.kt
+++ b/src/main/kotlin/com/replaymod/gradle/preprocess/PreprocessPlugin.kt
@@ -67,7 +67,7 @@ class PreprocessPlugin : Plugin<Project> {
             val reverseMappings = (inheritedLink != null) != mappingFileInverted
             val inherited = parent.evaluationDependsOn(inheritedNode.project)
 
-            project.the<SourceSetContainer>().configureEach {
+            project.the<SourceSetContainer>().configureEach { try {
                 val inheritedSourceSet = inherited.the<SourceSetContainer>()[name]
                 val cName = if (name == "main") "" else name.capitalize()
                 val overwritesKotlin = project.file("src/$name/kotlin").also { it.mkdirs() }
@@ -136,7 +136,7 @@ class PreprocessPlugin : Plugin<Project> {
                 }
                 project.tasks["process${cName}Resources"].dependsOn(preprocessResources)
                 resources.setSrcDirs(listOf(overwriteResources, preprocessResources.map { generatedResources }))
-            }
+            } catch (_: Throwable) {}}
 
             project.afterEvaluate {
                 val prepareTaskName = "prepareMappingsForPreprocessor"
