From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jab125 <67534807+Jab125@users.noreply.github.com>
Date: Sun, 4 Jun 2023 00:30:20 +1000
Subject: [PATCH] Bump version to 0.2.1


diff --git a/build.gradle.kts b/build.gradle.kts
index ec4dc79a6a53884d531ece48772aaf9fbdffc985..aee382a2543f70323179894e336f2c6e8d03e38b 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -21,7 +21,7 @@ plugins {
 }
 
 group = "com.jab125.preprocessor" // this is a fork
-version = "0.2.0"
+version = "0.2.1"
 
 val kotestVersion: String by project.extra
 
